app/org/sagebionetworks/bridge/dynamodb/DynamoNotificationRegistrationDao.java

```diff
@@ -0,0 +1,192 @@
+import com.amazonaws.services.sns.model.CreatePlatformEndpointRequest;
+import com.amazonaws.services.sns.model.CreatePlatformEndpointResult;
+import com.amazonaws.services.sns.model.DeleteEndpointRequest;
+import com.amazonaws.services.sns.model.GetEndpointAttributesRequest;
+import com.amazonaws.services.sns.model.GetEndpointAttributesResult;
+import com.amazonaws.services.sns.model.InvalidParameterException;
+import com.amazonaws.services.sns.model.NotFoundException;
+import com.amazonaws.services.sns.model.SetEndpointAttributesRequest;
+import com.google.common.collect.Maps;
+
+@Component
+public class DynamoNotificationRegistrationDao implements NotificationRegistrationDao {
+
+    private static final String CUSTOM_USER_DATA = "CustomUserData";
+    private static final String ENABLED = "Enabled";
+    private static final String TOKEN = "Token";
+
+    private DynamoDBMapper mapper;
+    
+    private AmazonSNSClient snsClient;
+    
+    @Resource(name = "notificationRegistrationMapper")
+    final void setNotificationRegistrationMapper(DynamoDBMapper mapper) {
+        this.mapper = mapper;
+    }
+    
+    @Resource(name = "snsClient")
+    final void setSnsClient(AmazonSNSClient snsClient) {
+        this.snsClient = snsClient;
+    }
+    
+    @Override
+    public List<NotificationRegistration> listRegistrations(String healthCode) {
+        checkNotNull(healthCode);
+        
+        DynamoNotificationRegistration hashKey = new DynamoNotificationRegistration();
+        hashKey.setHealthCode(healthCode);
+        
+        DynamoDBQueryExpression<DynamoNotificationRegistration> query =
+                new DynamoDBQueryExpression<DynamoNotificationRegistration>().withHashKeyValues(hashKey);
+
+        // This will force all records to be loaded if there's more than a page, in any non-pathological 
+        // account there will be half a dozen, tops
+        return mapper.query(DynamoNotificationRegistration.class, query)
+                .stream().collect(Collectors.toList());
+    }
+
+    @Override
+    public NotificationRegistration getRegistration(String healthCode, String guid) {
+        checkNotNull(healthCode);
+        checkNotNull(guid);
+        
+        DynamoNotificationRegistration hashKey = new DynamoNotificationRegistration();
+        hashKey.setHealthCode(healthCode);
+        hashKey.setGuid(guid);
+        
+        NotificationRegistration registration = mapper.load(hashKey);
+        if (registration == null) {
+            throw new EntityNotFoundException(NotificationRegistration.class);
+        }
+        return registration;
+    }
+
+    @Override
+    public NotificationRegistration createRegistration(String platformARN, NotificationRegistration registration) {
+        checkNotNull(platformARN);
+        checkNotNull(registration);
+        checkNotNull(registration.getHealthCode());
+        checkNotNull(registration.getDeviceId());
+        checkNotNull(registration.getOsName());
```

Comment:
As I'm building this, it doesn't seem that necessary, but I am just recording locally the device identifier and platform in case we need it for something. Otherwise we can't easily migrate later. The osName tells me whether the device identifier is really a device token (iOS) or a registrationId (Android).